src/shell/shell_cmds.cpp

```diff
@@ -608,6 +614,61 @@ static std::vector<int> calc_column_widths(const std::vector<int> &word_widths,
 	return col_widths;
 }
 
+char buffer[15] = {0};
+char *format_date(uint16_t year, uint8_t month, uint8_t day)
+{
+	char formatstring[6];
+	char date_format = dos.tables.country[0];
+	char date_separator = dos.tables.country[11];
+	sprintf(formatstring,
+	        date_format == 1 ? "D%cM%cY"
+	                         : (date_format == 2 ? "Y%cM%cD" : "M%cD%cY"),
+	        date_separator, date_separator);
+	Bitu bufferptr = 0;
+	for (Bitu i = 0; i < 5; i++) {
+		if (i == 1 || i == 3) {
+			buffer[bufferptr] = formatstring[i];
+			bufferptr++;
+		} else {
+			if (formatstring[i] == 'M')
+				bufferptr += (Bitu)sprintf(buffer + bufferptr,
+				                           "%02u", month);
+			if (formatstring[i] == 'D')
+				bufferptr += (Bitu)sprintf(buffer + bufferptr,
+				                           "%02u", day);
+			if (formatstring[i] == 'Y')
+				bufferptr += (Bitu)sprintf(buffer + bufferptr,
+				                           "%04u", year);
+		}
+	}
+	return buffer;
+}
+
+char *format_time(Bitu hour, Bitu min, Bitu sec, Bitu msec, bool full = false)
+{
+	Bitu fhour = hour;
+	static char retBuf[14];
+	char ampm[3] = "";
+	char time_format = dos.tables.country[17];
+	if (!(time_format & 1)) { // 12 hour notation?
+		if (hour != 12)
+			hour %= 12;
+		strcpy(ampm, hour != 12 && hour == fhour ? "am" : "pm");
+		if (!full)
+			*(ampm + 1) = 0;
+	}
+	char time_separator = dos.tables.country[13];
+	char decimal_separator = dos.tables.country[9];
+	if (full)
+		sprintf(retBuf, "%u%c%02u%c%02u%c%02u%s", (unsigned int)hour,
+		    time_separator, (unsigned int)min, time_separator, (unsigned int)sec,
+		    decimal_separator, (unsigned int)msec, ampm);
+	else
+		sprintf(retBuf, "%2u%c%02u%s", (unsigned int)hour,
+		        time_separator, (unsigned int)min, ampm);
+	return retBuf;
```

Comment:
Please add an intro comment showing an example output formatted time string.

The body is not easily readable (need more comments - specifically for (time_format & 1?) andde-referencing an (ampm + 1) pointer. 

For all the variables that aren't changed, please const them.

Please also use the camel_case convention in the maintenance wiki, instead of `retBuf`. 
 
Avoid bitu.

Reply:
Done.